### PR TITLE
Add Product Draft UUID routes

### PR DESCRIPTION
**Pull request by LevFlavien**

### CODE REVIEW

File: spec/Api/ProductDraftUuidApiSpec.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        $this->shouldHaveType(ProductDraftUuidApi::class);
+        $this->shouldImplement(ProductDraftUuidApiInterface::class);
+        $this->shouldImplement(GettableResourceInterface::class);
+    }
+
+    function it_gets_a_product_draft($resourceClient)
+    {
+        $draft = [
+            'uuid' => '944ca210-d8e0-4c57-9529-741e17e95c8d',
```
Comment (tomglvng) [inline spec/Api/ProductDraftUuidApiSpec.php:33]: I think you still should return an identifier
`'identifier' => 'foo',`

File: spec/Api/ProductDraftUuidApiSpec.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            'values' => [],
+            'created' => 'this is a date formatted to ISO-8601',
+            'updated' => 'this is a date formatted to ISO-8601',
+            'associations' => [],
+            'metadata' => [
+                'workflow_status' => 'draft_in_progress',
+            ],
+        ];
+
+        $resourceClient->getResource(ProductDraftUuidApi::PRODUCT_DRAFT_UUID_URI, ['foo'])->willReturn($draft);
```
Comment (tomglvng) [inline spec/Api/ProductDraftUuidApiSpec.php:48]:
```suggestion
        $resourceClient->getResource(ProductDraftUuidApi::PRODUCT_DRAFT_UUID_URI, ['944ca210-d8e0-4c57-9529-741e17e95c8d''])->willReturn($draft);
```

File: spec/Api/ProductDraftUuidApiSpec.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        $this->get('foo')->shouldReturn($draft);
```
Comment (tomglvng) [inline spec/Api/ProductDraftUuidApiSpec.php:50]:
```suggestion
        $this->get('944ca210-d8e0-4c57-9529-741e17e95c8d'')->shouldReturn($draft);
```

File: src/Api/ProductDraftUuidApi.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license https://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class ProductDraftUuidApi implements ProductDraftUuidApiInterface
+{
+    const PRODUCT_DRAFT_UUID_URI = '/api/rest/v1/products-uuid/%s/draft';
+    const PRODUCT_PROPOSAL_UUID_URI = '/api/rest/v1/products-uuid/%s/proposal';
+
+    /** @var ResourceClientInterface */
+    protected $resourceClient;
```
Comment (tomglvng) [inline src/Api/ProductDraftUuidApi.php:21]:
```suggestion
    protected ResourceClientInterface $resourceClient;
```
And remove comments (do the same for all attributes)

**Review by tomglvng — APPROVED** [review]
GTM after fixes

File: spec/Api/ProductDraftUuidApiSpec.php (hunk 2 above)
Comment (LevFlavien) [reply spec/Api/ProductDraftUuidApiSpec.php:48]: done

File: src/Api/ProductDraftUuidApi.php (hunk 4 above)
Comment (LevFlavien) [reply src/Api/ProductDraftUuidApi.php:21]: done

File: spec/Api/ProductDraftUuidApiSpec.php (hunk 3 above)
Comment (LevFlavien) [reply spec/Api/ProductDraftUuidApiSpec.php]: done

File: src/Api/ProductDraftUuidApi.php (hunk 4 above)
Comment (LevFlavien) [reply src/Api/ProductDraftUuidApi.php:21]: refactored

File: spec/Api/ProductDraftUuidApiSpec.php (hunk 1 above)
Comment (LevFlavien) [reply spec/Api/ProductDraftUuidApiSpec.php:33]: discussed: api spec doesn't mention `identifier` property

**Pull request merged**